### DISCUSSION TITLE
Simplifying LazyVim package management and NeoVim customization for Emacs user

### DISCUSSION THREAD

**Discussion by jasalt** [LazyVim/LazyVim#2094] [General]
I migrated from VSCode as main code editor (PHP/Python) to LazyVim/Neovim earlier this year after a failed attempt to use LSP, DAP and tree-sitter in Emacs (28) that I have used previously for notes, writing etc. Got it up and running relatively fast (~week or two), finished some work with it happily and got much more accustomed to general vi bindings which has been particularly helpful when working with servers over SSH.

NeoVim customization however still mostly remains a mystery to me and I am pretty much just copy pasting code found from online, editing config file on separate editor process and restarting editor numerous times to see if I find correct syntax or not, eg. beginner problems.

NeoVim's Lua layer and it's parallel life with Vimscript are confusing to me while in Emacs config and core code are simply e-lisp all the way so that things like jump to definitions and config code runtime evaluation are working more consistent out-of-box. NeoVim self documentation facilities are also not yet clear to me, though I am guessing there is probably something similar to "C-h" binding help commands etc. on Emacs side.

Lately hitting wall customizing some small things in NeoVim, I started rewriting .emacs.d for Emacs 29 which has LSP and tree-sitter included to core, and had more success with it now getting it working from previous attempts writing code with IDE features on it (knock wood).

So I would appreciate some kind of intro to NeoVim for Emacs (29) users if there is something like that out there.

AFAIK it seems like it would probably be better to start with NeoVim using minimal Kickstart config framework and learn it more ground up, but it seems like have enough passion for that at the moment.

**Comment by dpetka2001** [comment]
That would have to be made by someone who is familiar with both Emacs and Neovim and would also be better to be included in the Neovim website, if the core devs would even be interested in accommodating such a feat. Unfortunately, the best way to get acquainted with Neovim is to spend time configuring it by yourself and search things that you have trouble achieving for your workflow. Nobody will be able to do that for you. I've been using Neovim just for a year before transitioning to Lazyvim (was using a `init.vim` with some lua files taken by TjDevries to achieve what I wanted) and in the beginning I definitely had trouble getting adjusted to the new way of configuring things. I just kept doing it. That's all. No secret magic required. If you don't have enough time to invest to learn how to do it, it's perfectly ok and nobody would blame you. Just stick to what makes you feel more comfortable with your workflow. The thing I'd usually suggest for new users, who already have a lot on their plate, is to see it as a side project and take it slow, so that they can understand why they're doing some things and not blindly copy/paste. In the end, it's all up to you.
